library/std/src/fs.rs

```diff
@@ -795,67 +796,99 @@ impl Write for File {
     }
 }
 #[stable(feature = "rust1", since = "1.0.0")]
-impl Seek for File {
+impl Seek for &File {
     fn seek(&mut self, pos: SeekFrom) -> io::Result<u64> {
         self.inner.seek(pos)
     }
 }
+
 #[stable(feature = "rust1", since = "1.0.0")]
-impl Read for &File {
+impl Read for File {
```

Comment:
Is this change correct wrt the resolver?

Reply:
`Read` (etc.) are still implemented for `File` **and** `&File`. No changes made to the outside. I just moved the implementation into the `&File` impl because I can forward the `File` impl to the `&File` one.